nutils/evaluable.py

```diff
@@ -2340,27 +2361,32 @@ def evalf(self, index):
 
 class Eig(Evaluable):
 
-  __slots__ = 'symmetric', 'func'
+  __slots__ = 'symmetric', 'func', 'w_dtype', 'vt_dtype'
```

Comment:
We are not very consistent with making attributes public or private, but I would say these type attributes qualify for being private. If we need to obtain the types of the eigenvalues or the eigenvectors outside of this class, we can easily do so via the `ArrayFromTuple` s.